update/admin/product-collection-group

wychoong opened the pull request:
Cosmetic enhancement
resolves #424

**Collection page**
- show collection's breadcrumbs and group name as label
- fix z-index of search result and button
after
![image](https://user-images.githubusercontent.com/67364036/181456892-a464396f-c6c8-4dd7-aa45-adc189835695.png)

before
![image](https://user-images.githubusercontent.com/67364036/181457468-862e2300-212e-4991-9447-b34efb8be3b4.png)

**Product page**
- show group name as label
after
![image](https://user-images.githubusercontent.com/67364036/181457818-d00760ac-ded8-4a4f-860b-69e88a667933.png)

before
![image](https://user-images.githubusercontent.com/67364036/181423581-3aa6f0b6-4cc5-4c3e-b391-f1a6a896c736.png)

wychoong commented:
resolves #424

markmead commented on packages/admin/resources/views/livewire/components/collections/collection-groups/show.blade.php:
```diff
@@ -115,18 +115,21 @@ class="w-4" />
                                            )" />
                         @if ($showCollectionSearchResults)
-                            <div class="absolute w-full overflow-y-scroll bg-white rounded-b shadow max-h-64">
+                            <div class="absolute w-full overflow-y-scroll bg-white rounded-b shadow max-h-64 z-10">
```
What's the need for the `z-10` here? I've not come across any index issues 🤔

wychoong replied:
its for the Move button, refer to this
![image](https://user-images.githubusercontent.com/67364036/181457468-862e2300-212e-4991-9447-b34efb8be3b4.png)

markmead replied:
Ah! In that case we don't need the `z-0` on the button, the `z-10` you added will be perfect!

wychoong replied:
sure, the `z-0` is more of a safety

markmead commented on packages/admin/resources/views/livewire/components/collections/collection-groups/show.blade.php:
```diff
@@ -115,18 +115,21 @@ class="w-4" />
 
-                                        <x-hub::icon ref="chevron-right"
+                                        @foreach($collection->ancestors as $ancestor)
+                                            {{$ancestor->translateAttribute('name')}}
```
Let's wrap this in a `<span>` and keep the `text-gray-500`.

Could we add a space between `{{` and `}}` and a new line, please?

wychoong replied:
`text-gray-500` is for which element?

markmead commented on packages/admin/resources/views/livewire/components/collections/collection-groups/show.blade.php:
```diff
@@ -115,18 +115,21 @@ class="w-4" />
                                             type="button"
                                             wire:click.prevent="setMoveTarget('{{ $collection->id }}')">
-                                        <span class="text-gray-500">
+                                        <span class="rounded mr-2 px-2 bg-blue-50 text-blue-700 dark:bg-gray-800 dark:text-white;">
```
Don't need `mr-2` as we are already using `gap-1.5` on the parent.

Let's remove the `dark:` classes for now and we can revisit once we start the dark mode PR.

wychoong replied:
the `mr-2` is required, perhaps `gap-1.5` doesn't apply to `<span>`

markmead replied:
It works fine, I have it running locally, and looks OK.

wychoong replied:
it looks like this on mine
![image](https://user-images.githubusercontent.com/67364036/181475961-1ed37164-c088-431a-b1f1-9a917852da7e.png)

will remove it anyways

markmead replied:
Might need to run `yarn watch`.

markmead commented on packages/admin/resources/views/livewire/components/collections/collection-groups/show.blade.php:
```diff
@@ -115,18 +115,21 @@ class="w-4" />
+                            <div class="absolute w-full overflow-y-scroll bg-white rounded-b shadow max-h-64 z-10">
                                 @foreach ($this->searchedCollections as $collection)
-                                    <button class="flex gap-1.5 items-center w-full px-4 py-2 text-sm text-left hover:bg-gray-100"
+                                    <button class="flex gap-1.5 items-center w-full px-4 py-2 text-gray-800 text-sm text-left hover:bg-gray-100"
```
Do we need `text-gray-800` all the text colors have been set on the child elements?

markmead commented on packages/admin/resources/views/livewire/components/collections/collection-groups/show.blade.php:
```diff
@@ -148,6 +151,7 @@ class="w-4 h-4 text-gray-500"
                 <x-slot name="footer">
                     <x-hub::button type="button"
                                    wire:click="moveCollection"
+                                   class="z-0"
```
Do you have an example of why we need this (and the other index change)?

wychoong replied:
this is related `z-10` and consider resolved

markmead commented on packages/admin/resources/views/partials/products/editing/collections.blade.php:
```diff
@@ -18,6 +18,7 @@
             @endif
             <div class="flex grow">
```
I'll pull this down and make some changes, I think the existing code could be better. Your changes look good!

wychoong replied:
awesome, great that it can be accepted  😁

markmead commented:
OK, I've made some changes to my PR based on the work you've done. I'm happy with how that looks now.

Do you want to take that and apply it to [the product page](https://github.com/getcandy/getcandy/pull/431/files/18ffd0b07fd907e0b7b076a6adcb9629a3667a00#diff-70e06115e3036898f08d7571fc325824dae3318b0831b00759af0d8e04b93165)?

wychoong commented:
> OK, I've made some changes to my PR based on the work you've done. I'm happy with how that looks now.
> 
> Do you want to take that and apply it to [the product page](https://github.com/getcandy/getcandy/pull/431/files/18ffd0b07fd907e0b7b076a6adcb9629a3667a00#diff-70e06115e3036898f08d7571fc325824dae3318b0831b00759af0d8e04b93165)?
> 
> Once done we can merge this PR into mine and get it merged! 🥳

i updated the collection group label styling

and also moved the breadcrumb into same line with the collection, not sure will this be wanted so i commit it separately
![image](https://user-images.githubusercontent.com/67364036/181489318-8af07b3e-e391-480f-b391-047d76b4d694.png)

markmead commented:
Yeah, that looks good to me! 

Looks like we have some sneaky `app.css` and `app.js` files being committed.

wychoong commented:
> Yeah, that looks good to me!
> 
> Looks like we have some sneaky `app.css` and `app.js` files being committed.

in the previous 2 lines style have issue on the whitespace in `<strong>` so I had to use `inline-block` class. 

What is the asset build process for getCandy? So next time I’ll keep in mind to exclude the files 😅

alecritson commented:
Hey @wychoong Seems normal to have css js files committed if there has been changes, since these are published to the developers laravel app

wychoong commented:
@markmead i have fixed the test for the changes, it's all pass now on my local

markmead commented:
Awesome work @wychoong! 🥳 

Looks good to me! Any thoughts @alecritson?

alecritson commented:
@markmead Happy if you are! Let's get it in
